modules/backend/widgets/form/partials/_field_switch.htm

```diff
@@ -12,15 +12,15 @@
     value="0"
     <?= $this->previewMode ? 'disabled="disabled"' : '' ?>>
 
-<label class="custom-switch" <?= $this->previewMode ? 'onclick="return false"' : '' ?>>
+<label class="custom-switch" <?= ($this->previewMode || $field->readOnly) ? 'onclick="return false"' : '' ?>>
```

Comment:
If this is the only change that you need to make to the switch widget, then a better solution is to force readOnly = true to set previewMode = true

Reply:
That's a good approach, I didn't want to touch any of the current attributes in order to prevent propagation of undesired behaviours later.

Reply:
See https://github.com/octobercms/october/commit/5f91c45f7997571290b3cbc1307a69b642d6c6e6#diff-24b825f292c5e6c4f2ce1bf5d6204b68 for an example of how @daftspunk implemented `disabled` support on the FileUpload form widget. It has a precedent so that should be fine.